components/database/mariadb-101/mariadb-101.p5m

```diff
@@ -0,0 +1,317 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2013 Alexander Pyhalov.  All rights reserved.
+# Copyright 2016-2017, Wiselabs Software Ltda - http://www.wiselabs.com.br . All rights reserved.
+#
+
+set name=pkg.fmri \
+    value=pkg:/database/mariadb-101@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary \
+    value="MariaDB 10.1 Database Management System"
+set name=info.classification \
+    value=org.opensolaris.category.2008:Development/Databases
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+<transform file path=usr.*/man/.+ -> default mangler.man.stability committed>
+
+dir path=var/mariadb group=mysql mode=0700 owner=mysql
+dir path=var/mariadb/10.1 group=mysql mode=0700 owner=mysql
+dir path=var/mariadb/10.1/data group=mysql mode=0700 owner=mysql
+file files/my-medium.cnf path=etc/mariadb/10.1/my.cnf \
+    mode=0644 preserve=renamenew
+file files/auth_attr path=etc/security/auth_attr.d/mariadb-101
+file files/prof_attr path=etc/security/prof_attr.d/mariadb-101
+file files/mariadb_101.xml \
+    path=lib/svc/manifest/application/database/mariadb_101.xml
+file files/mariadb_101 path=lib/svc/method/mariadb_101 mode=0555
+
+license mariadb-101.license license=GPLv2
+
+link path=etc/mysql/my.cnf target=../mariadb/10.1/my.cnf mediator=mysql mediator-version=10.1 mediator-implementation=mariadb
+link path=usr/mariadb/10.1/bin/64 target=$(MACH64)
+link path=usr/mysql/bin target=../mariadb/10.1/bin mediator=mysql mediator-version=10.1 mediator-implementation=mariadb
+link path=usr/mysql/docs target=../mariadb/10.1/docs mediator=mysql mediator-version=10.1 mediator-implementation=mariadb
+link path=usr/mysql/man target=../mariadb/10.1/man mediator=mysql mediator-version=10.1 mediator-implementation=mariadb
+link path=usr/mysql/share target=../mariadb/10.1/share mediator=mysql mediator-version=10.1 mediator-implementation=mariadb
+link path=var/mysql/data target=../mariadb/10.1/data mediator=mysql mediator-version=10.1 mediator-implementation=mariadb
+
+depend fmri=database/mysql-common type=require
+depend fmri=database/mariadb-101/client@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION) type=require
+
+dir  path=usr/mariadb
```

Comment:
it's not needed at all if you deliver files in directories.
It's necessary only for directories with special permissions, directories which are target for symlinks and empty directories which you want to deliver.